rewrite-java/src/main/java/org/openrewrite/java/search/IsLikelyTest.java

```diff
@@ -0,0 +1,69 @@
+/*
+ * Copyright 2023 the original author or authors.
+ * <p>
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ * <p>
+ * https://www.apache.org/licenses/LICENSE-2.0
+ * <p>
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.openrewrite.java.search;
+
+import org.openrewrite.*;
+import org.openrewrite.java.JavaIsoVisitor;
+import org.openrewrite.java.marker.JavaSourceSet;
+import org.openrewrite.java.tree.JavaSourceFile;
+import org.openrewrite.marker.SearchResult;
+
+import java.util.regex.Pattern;
+
+@Incubating(since = "7.36.0")
+public class IsLikelyTest extends Recipe {
+    @Override
+    public String getDisplayName() {
+        return "Find sources that are likely tests";
+    }
+
+    @Override
+    public String getDescription() {
+        return "Sources that contain indicators of being, or being exclusively for the use in tests. " +
+                "This recipe is not exhaustive, but is intended to be a good starting point for finding test sources. " +
+                "Looks at the source set name, and types in use; for example looks for uses of JUnit & TestNG annotations/assertions.";
+    }
+
+    @Override
+    protected TreeVisitor<?, ExecutionContext> getVisitor() {
+        return Applicability.or(
+                new HasSourceSet("test").getVisitor(),
+                new HasSourceSetIntegrationTestVisitor<>(),
+                new UsesType<>("org.junit..*"), // Covers both JUnit 4 and 5
+                new UsesType<>("org.testng..*"),
+                new UsesType<>("org.hamcrest..*"),
+                new UsesType<>("org.mockito..*"),
+                new UsesType<>("org.powermock..*"),
+                new UsesType<>("org.assertj..*"),
+                new UsesType<>("spock.lang..*")
+        );
+    }
+
+    private static class HasSourceSetIntegrationTestVisitor<P> extends JavaIsoVisitor<P> {
+        private static final Pattern INTEGRATION_TEST_PATTERN =
+                Pattern.compile("integ.*test", Pattern.CASE_INSENSITIVE);
+
+        @Override
+        public JavaSourceFile visitJavaSourceFile(JavaSourceFile cu, P p) {
+            if (cu.getMarkers().findFirst(JavaSourceSet.class)
+                    .filter(s -> INTEGRATION_TEST_PATTERN.matcher(s.getName()).matches())
```

Comment:
This may be better to check for source sets that **contain** "test". 

In Gradle, they make it really easy to create additional test suites, but the convention is that it should be <*suite*>Test. (ex: compatibilityTest, functionalTest, integrationTest, etc).

Reply:
Good idea, updated. 👍